File: VB/DialogsDemo.Wpf/Helpers/ThemedWindowService.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports DevExpress.Mvvm.UI
Imports DevExpress.Xpf.Core

Namespace DialogsDemo
    Public Interface IThemedWindowService
        Function Show(ByVal dataContext As Object) As Boolean
        Function Hide() As Boolean
        ReadOnly Property IsWindowOpened() As Boolean
    End Interface
    Public Class ThemedWindowService
        Inherits ServiceBase
        Implements IThemedWindowService

        Private themedWindow As ThemedWindow
        Public Property WindowStyle() As Style
        Public Sub New()
        End Sub

        Private Sub OnThemedWindowClosed(ByVal sender As Object, ByVal e As EventArgs)
            IsWindowOpened = False
            RemoveHandler themedWindow.Closed, AddressOf OnThemedWindowClosed
            themedWindow = Nothing
        End Sub

        Public Function Show(ByVal dataContext As Object) As Boolean Implements IThemedWindowService.Show
            If IsWindowOpened Then
                Return False
            End If
            IsWindowOpened = True
            themedWindow = New ThemedWindow()
            AddHandler themedWindow.Closed, AddressOf OnThemedWindowClosed
            themedWindow.DataContext = dataContext
            themedWindow.Style = WindowStyle
            themedWindow.Show()
            Return True
        End Function

        Public Function Hide() As Boolean Implements IThemedWindowService.Hide
            If Not IsWindowOpened Then
                Return False
            End If
            themedWindow.Close()
            Return True
        End Function

        Private ReadOnly Property IThemedWindowService_IsWindowOpened() As Boolean Implements IThemedWindowService.IsWindowOpened
            Get
                Return IsWindowOpened
            End Get
        End Property
        Private privateIsWindowOpened As Boolean
        Public Property IsWindowOpened() As Boolean
            Get
                Return privateIsWindowOpened
            End Get
            Private Set(ByVal value As Boolean)
                privateIsWindowOpened = value
            End Set
        End Property
    End Class
End Namespace

File: CS/DialogsDemo.Wpf/Helpers/ThemedWindowService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using DevExpress.Mvvm.UI;
using DevExpress.Xpf.Core;

namespace DialogsDemo {
    public interface IThemedWindowService {
        bool Show(object dataContext);
        bool Hide();
        bool IsWindowOpened { get; }
    }
    public class ThemedWindowService : ServiceBase, IThemedWindowService {
        ThemedWindow themedWindow;
        public Style WindowStyle { get; set; }
        public ThemedWindowService() {            
        }

        void OnThemedWindowClosed(object sender, EventArgs e) {
            IsWindowOpened = false;
            themedWindow.Closed -= OnThemedWindowClosed;
            themedWindow = null;            
        }

        public bool Show(object dataContext) {
            if (IsWindowOpened)
                return false;
            IsWindowOpened = true;
            themedWindow = new ThemedWindow();
            themedWindow.Closed += OnThemedWindowClosed;            
            themedWindow.DataContext = dataContext;
            themedWindow.Style = WindowStyle;
            themedWindow.Show();
            return true;
        }

        public bool Hide() {
            if (!IsWindowOpened)
                return false;
            themedWindow.Close();            
            return true;
        }

        public bool IsWindowOpened { get; private set; }
    }
}
